param/__init__.py

```diff
@@ -1014,6 +1014,9 @@ def _validate(self, val):
                              (self.name,len(val),self.length))
 
 
+    def _deserialize(self, value):
```

Comment:
Can we make serialize/deserialize public methods? I will be using them heavily. Also can they be classmethods?

Reply:
Applied this suggestion in 56d5089 and edadd08 though it is worth nothing the names aren't really right as neither serialize nor deserialize actually work with strings (i.e the serialized form).

`serialize_python_value` and `deserialize_python_value` would be more accurate names but also ugly.

Reply:
Bokeh calls it `transform` and `from_json` (even though it's already a Python object by that point). The former would work but the latter is clearly not appropriate, they also aren't symmetric like `serialize` and `deserialize` so I'm okay keeping the names.

Reply:
ir_serialize and ir_deserialize?

Reply:
Used this suggestion in a7a4449